grails-app/services/org/broadinstitute/orsp/QueryService.groovy

```diff
@@ -349,11 +351,35 @@ class QueryService implements Status {
                 .setString('consentKey', consentKey)
                 .list()
         results.collect {
-            sampleInfo.put(it, StorageDocument.findAllByConsentCollectionLinkId(it.id))
+            consentCollectionIds.add(it.id)
+            consentCollectionLinkList.add(it)
+        }
+        List <StorageDocument> storageDocuments = findAllDocumentsBySampleCollectionId(consentCollectionIds)
+        consentCollectionLinkList.collect {
+            sampleInfo.put(it,
+                    storageDocuments.findAll{
```

Comment:
This is also a nested query ... i.e., you're executing 1 query per item in `consentCollectionLinkList`. I think you just moved the problem to a new location in the code rather than fixing the problem.

Reply:
Oh! eitherway these are not mysql queries if that's what you meant here. Made changes to remove findAll.